mautic/mautic
# Form will throw the uncaught errors if form is nor valid.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ 5.x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

**Issue**: 
When attempting to upload a theme ZIP file larger than the allowed size set in the `upload_max_filesize` PHP setting, clicking the "Install" button does not result in feedback or action. An error message should be displayed to inform the user that the file exceeds the allowed limit.

#### Steps to Reproduce:

1. Update your PHP configuration (`php.ini`) to set the `upload_max_filesize` to **1M**.
    - Make sure to restart the server after making this change. (Please see [DDEV's docs](https://ddev.readthedocs.io/en/stable/users/extend/customization-extendibility/#custom-php-configuration-phpini))
2. Navigate to the theme upload page at `/s/themes`.
3. Click on the "Choose" button and select a theme ZIP file larger than 1MB.
4. Click on the "Install" button.
5. **Expected behavior**: Nothing happens, and no message is displayed.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Update your PHP configuration (`php.ini`) to set the `upload_max_filesize` to **1M**.
    - Please make sure you restart the server after making this change. (Please see [DDEV's docs](https://ddev.readthedocs.io/en/stable/users/extend/customization-extendibility/#custom-php-configuration-phpini))
3. Navigate to the theme upload page at `/s/themes`.
4. Click the "Choose" button and select a theme ZIP file larger than 1MB.
5. Click on the "Install" button.
6. **Expected behavior**: The form should display the error message `ERROR: The file is too large.` indicating that the selected file exceeds the allowed upload size.

NOTE: 
- The changes in `php.ini` are meant for testing only. Do not lower the value of the `upload_max_filesize` config setting.

#### Extras,
- Screencast

https://github.com/user-attachments/assets/c192bf33-9136-49d5-bb04-693e737b3c57

- Compressed zip
[mautic-wide.zip](https://github.com/user-attachments/files/17264739/mautic-wide.zip)
